mbed_targets/_internal/target_attribute_hierarchy_parsers/accumulating_attribute_parser.py

```diff
@@ -13,18 +13,31 @@
 from typing import Dict, List, Any, Deque
 
 ACCUMULATING_ATTRIBUTES = ["extra_labels", "macros", "device_has", "features", "components"]
-ALL_ACCUMULATING_ATTRIBUTES = ACCUMULATING_ATTRIBUTES + [
-    "extra_labels_remove",
-    "extra_labels_add",
-    "macros_remove",
-    "macros_add",
-    "device_has_remove",
-    "device_has_add",
-    "features_remove",
-    "features_add",
-    "components_remove",
-    "components_add",
-]
+ACCUMULATING_ATTRIBUTES_MODIFIERS = {"add": "_add", "remove": "_remove"}
+
+
+def _modifying_attribute_name(attribute: str, modifying_action: str) -> str:
+    return f"{attribute}{ACCUMULATING_ATTRIBUTES_MODIFIERS[modifying_action]}"
+
+
+def _list_all_accumulating_attributes_and_modifiers() -> List[str]:
+    """Provides a list of all accumulating attributes and their modifying attributes.
+
+    For example, the accumulating attribute "extra_labels" can be modified by "_add" and
+    "_remove" to make "extra_labels_add" and "extra_labels_remove".
+
+    Returns:
+        A list of all the accumulating attributes and their modifiers.
+    """
+    all_accumulating_attributes = ACCUMULATING_ATTRIBUTES.copy()
+    for modifier_action in ACCUMULATING_ATTRIBUTES_MODIFIERS.keys():
+        all_accumulating_attributes.extend(
+            [_modifying_attribute_name(attribute, modifier_action) for attribute in ACCUMULATING_ATTRIBUTES]
+        )
+    return all_accumulating_attributes
+
+
+ALL_ACCUMULATING_ATTRIBUTES = _list_all_accumulating_attributes_and_modifiers()
```

Comment:
Less code, but unsure if readable. `itertools` is quite powerful. Maybe somebody has a better idea.
```
ALL_ACCUMULATING_ATTRIBUTES = ACCUMULATING_ATTRIBUTES + [f"{attr}{suffix}" for attr, suffix in itertools.product(ACCUMULATING_ATTRIBUTES, ["_add", "_remove"])]
```

Also, should these contants be `set`s?

Reply:
Alternative:
```
_WITH_ADD = [f"{attr}_add" for attr in ACCUMULATING_ATTRIBUTES]
_WITH_REMOVE = ...
ALL = ACCUMULATING_ATTRIBUTES + _WITH_ADD + _WITH_REMOVE
```

Reply:
In the sets v tuples I think I prefer tuples. Doesn't really matter if there are duplicates but I want it to be immutable I think.

I like the idea of using itertools. I will have a play around and see how it looks.

Reply:
45ed7ec